web/source/keymanweb.js

```diff
@@ -1318,7 +1317,25 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
     keymanweb.updateInput=function(x){};
   
     // End of I3363 (Build 301) additions
-    
+
+    /**
+     * Function     resetContext
+     * Scope        Public
+     * Description  Revert OSK to default layer and clear any deadkeys and modifiers
+     */
+    keymanweb['resetContext'] = keymanweb.resetContext = function()
+    {
+      osk.layerId = 'default';
+      osk.layerIndex = 0;
```

Comment:
No need to bother with the `osk.layerIndex = 0;` line - that's automatically managed by osk._Show() once `osk.layerId` is set.

Also, check the file formatting!  I believe our official policy says the opening brace should be at the end of the previous line.

Reply:
hmm. So for this set up:

1. Alt-Ctrl layer
2. `~`
3. Switch text field

The keyboard is staying in the Alt-Ctrl layer.  I'll have to see what I'm missing...